index.bs

```diff
@@ -8508,6 +8527,357 @@ worker=] algorithm:
+
+</div>
+
+<p>The following <dfn>table for cookie conversion</dfn>
+ defines the cookie concepts relevant to WebDriver BiDi,
+ how these are referred to in [[COOKIES]],
+ as well as what keys they map to in a [=serialized cookie=].
+
+<p>For informational purposes,
+ the table includes a brief non-normative description of the field.
+
+<table class=data>
+ <tr>
+  <th>Concept
+  <th>RFC 6265 Field
+  <th>JSON Key
+  <th>Description
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie name</dfn>
+  <td><code>name</code>
+  <td>"<code>name</code>"
+  <td>The name of the cookie.
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie value</dfn>
+  <td><code>value</code>
+  <td>"<code>value</code>"
+  <td>The cookie value.
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie path</dfn>
+  <td><code>path</code>
+  <td>"<code>path</code>"
+  <td>The cookie path.
+   Defaults to "<code>/</code>"
+   if omitted when [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie domain</dfn>
+  <td><code>domain</code>
+  <td>"<code>domain</code>"
+  <td>The domain the cookie is visible to.
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie secure only</dfn>
+  <td><code>secure-only-flag</code>
+  <td>"<code>secure</code>"
+  <td>Whether the cookie is a secure cookie.
+   Defaults to false if omitted when [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie HTTP only</dfn>
+  <td><code>http-only-flag</code>
+  <td>"<code>httpOnly</code>"
+  <td>Whether the cookie is an HTTP only cookie.
+   Defaults to false if omitted when [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie expiry time</dfn>
+  <td><code>expiry-time</code>
+  <td>"<code>expires</code>"
+  <td>When the cookie expires, specified in seconds since
+   [=Unix Epoch=]. Must not be set if omitted when
+   [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie same site</dfn>
+  <td><code>samesite</code>
+  <td>"<code>sameSite</code>"
+  <td>Whether the cookie applies to a SameSite policy.
+    Defaults to None if omitted when [=storage.setCookie|setting a cookie=].
+    Can be set to either <code>[=Lax=]</code> or <code>[=Strict=]</code>.
+ </tr>
+</table>
+
+<div algorithm>
+To <dfn>match cookie</dfn> given |stored cookie| and |filter|:
+
+1. For each |name| → |value| in |filter|:
+  1. If |name| is "<code>value</code>":
+    1. Set |value| to [=deserialize protocol bytes=] with |value|.
+  1. Let |field name| be the field name corresponding to the JSON key |name| in
+    the [=table for cookie conversion=].
+  1. If |stored cookie|[|field name|] does not equal |value|:
+    1. Return false.
+1. Return true.
+
+</div>
+
+<div algorithm>
+To <dfn>get matching cookies</dfn> given |cookie store| and |filter|:
+
+1. Let |cookies| be a new list.
+1. For each |stored cookie| in |cookie store|:
+  1. If [=match cookie=] with |stored cookie| and |filter| is true:
+    1. Append the result of [=serialize cookie=] given |stored cookie| to |cookies|.
+1. Return |cookies|.
```

Comment:
This could be expressed without an algorithm like "match cookie", but doing so would involve labeling a loop in order to continue from within a nested loop. Factoring the operation into two simpler algorithms seems preferable.